consensus/src/sync/live/block_queue/mod.rs

```diff
@@ -296,12 +299,75 @@ impl<N: Network> BlockQueue<N> {
         );
     }
 
+    /// Attempts to insert a block into the buffer.
+    /// This may fail due to the peer who propagated the block as given in the optional `pubsub_id` parameter
+    /// already having too many blocks buffered. Blocks without a pubsub_id will always be buffered.
+    ///
+    /// At most two block per peer per block height will be buffered leaving enough room for a bock and the
+    /// skip block making it obsolete. For benign peers that is sufficient, as they will never propagate
+    /// more than that.
+    ///
+    /// ## Returns
+    /// * `true` if the block was added to the buffer
+    /// * `false` otherwise
     fn insert_block_into_buffer(&mut self, block: Block, pubsub_id: Option<N::PubsubId>) -> bool {
```

Comment:
```suggestion
    fn insert_block_into_buffer(&mut self, block: Block, pubsub_id: Option<N::PubsubId>) -> Result<(), ()> {
```
`Result<(), ()>` is perfect for this, but clippy will be mad at you… Perhaps just `#[allow(clippy::result_unit_err)]`?

Reply:
I was considering `Result<bool, ()>` to 
`Ok(true)` if the block was added, 
`Ok(false)` if the block was already known 
`Err(())` if the block was not added due to buffering limitations
I decided that that level of granularity was not needed caller side, so I kept the `bool`. 
Why do you prefer `Result<(), ()>` over `bool`?

Reply:
I prefer `Result<(), ()>` over `bool` here, because I won't have to look at the documentation to know that `Ok(())` means that something succeeded, but I don't know if `true` or `false` means that the block was added. But it's more a question of taste, so it's not important to me to change this.